syspage.c

```diff
@@ -140,30 +144,30 @@ static int syspage_strAttr2ui(const char *str, unsigned int *attr)
 
 static void syspage_uiAttr2str(unsigned int attr, char *str)
 {
-	unsigned int i = 0;
-	unsigned int pos = 0;
+	size_t i, pos = 0;
 
 	for (i = 0; i < 32; ++i) {
 		switch (attr & (1 << i)) {
 			case mAttrRead:
-				str[pos++] = 'r';
+				str[pos] = 'r';
 				break;
 			case mAttrWrite:
-				str[pos++] = 'w';
+				str[pos] = 'w';
 				break;
 			case mAttrExec:
-				str[pos++] = 'x';
+				str[pos] = 'x';
 				break;
 			case mAttrShareable:
-				str[pos++] = 's';
+				str[pos] = 's';
 				break;
 			case mAttrCacheable:
-				str[pos++] = 'c';
+				str[pos] = 'c';
 				break;
 			case mAttrBufferable:
-				str[pos++] = 'b';
+				str[pos] = 'b';
 				break;
 		}
+		++pos;
 	}
 
 	str[pos] = '\0';
```

Comment:
Your implementation is not good.
Simple example:
`attr == 1; // mAttrRead`
`str == "uruuuuuuuuuuuuuuuuuuuuuuuuuuuuuu\0"`
where `u` means undefined
Leave it as it is now or change it to : 
```c
{
	char *p = str;

	if ((attr & mAttrRead) != 0) {
		*p++ = 'r';
	}

	if ((attr & mAttrWrite) != 0) {
		*p++ = 'w';
	}

	if ((attr & mAttrExec) != 0) {
		*p++ = 'x';
	}

	if ((attr & mAttrShareable) != 0) {
		*p++ = 's';
	}

	if ((attr & mAttrCacheable) != 0) {
		*p++ = 'c';
	}

	if ((attr & mAttrBufferable) != 0) {
		*p++ = 'b';
	}

	*p = '\0';
}
```

Reply:
I'll revert it back.